scripts/src/main/resources/scripts/command/java

```diff
@@ -96,10 +92,197 @@ function doCreate() {
+  fi
+
+  if [[ -e "${DEVON_IDE_HOME}/updates/quarkus.zip" ]]
+  then 
+    rm "${DEVON_IDE_HOME}/updates/quarkus.zip"
+  fi 
+  doDownload "${url}" "quarkus.zip" "${DEVON_IDE_HOME}/updates"
+
+  doExtract "${DEVON_IDE_HOME}/updates/quarkus.zip"
+  rm "${DEVON_IDE_HOME}/updates/quarkus.zip"
+
+  local groupIdFolders="${group//[.]/"/"}"
+  doCreateModernProjectStructure "${artifact}" "${PWD}" "${groupIdFolders}"
+  doMoveQuarkusFilesToModernStructure "${extractedFolderBasePath}/" "${PWD}/${artifact}/"
+
+  ## cleanup
+  rm -r "${extractedFolderBasePath}"
+}
+
+function doMoveQuarkusFilesToModernStructure() {
+  ## mv dockerfolder can be moved 
+  mv "${DEVON_IDE_HOME}/updates/extracted/${artifact}/src/main/docker" "${PWD}/${artifact}/src/main"
+  ## resoruces folder can be moved
+  mv "${DEVON_IDE_HOME}/updates/extracted/${artifact}/src/main/resources" "${PWD}/${artifact}/src/main"
+
+  if [ -d "${extractedFolderBasePath}/src/test/resources" ]
+  then 
+    mv "${extractedFolderBasePath}/src/test/resources" "${PWD}/${artifact}/src/test/"
+  fi
+
+  ##files=`eval find "${extractedFolderBasePath}/*" -type f`
+  restFiles=$(grep --with-filename '@Path' -r "${extractedFolderBasePath}/")
+  if [[ -n $restFiles ]]
+  then 
+    mv "${restFiles/:@Path*}" "${PWD}/${artifact}/src/main/java/${groupIdFolders}/${artifact}/rest/v1"
+  fi
+
+  testFiles=$(eval find '${extractedFolderBasePath}/src/test/*' -type f)
+  if [[ -n $testFiles ]]
+  then
+    mkdir -p "${PWD}/${artifact}/test/java/${groupIdFolders}/${artifact}/rest/v1"
+    mv "${testFiles}" "${PWD}/${artifact}/test/java/${groupIdFolders}/${artifact}/rest/v1" 
+  fi
+
+  projectFiles=$(eval find "${extractedFolderBasePath}/" -maxdepth 1 -type f)
+  if [[ -n $projectFiles ]]
+  then 
+    mv "${projectFiles}" "${DEVON_IDE_HOME}/updates/extracted/${artifact}/.mvn" "${PWD}/${artifact}/"
+  fi
+
+  otherFiles=$(eval find "${extractedFolderBasePath}/" -type f)
+  if [[ -n $otherFiles ]]
+  then 
+    mkdir -p "${PWD}/${artifact}/otherFiles"
+    doEchoAttention "We have found some files provided by Quarkus that couldn't be assigned to our project structure, we moved them into otherFiles folder and you have to relocate them by yourself"
+    mv "${otherFiles}" "${PWD}/${artifact}/otherFiles"
+  fi
+}
+
+
+#1 artifactId
+#2 path where the project will be created 
+#3 groupId
+function doCreateModernProjectStructure() {
+  local projectBasePath="${2}/${1}/src/main"
+  mkdir -p "${projectBasePath}" "${projectBasePath}/helm" "${projectBasePath}/java" "${projectBasePath}/java/${3}/general/domain/model" 
+  mkdir -p "${projectBasePath}/java/${3}/${1}/domain/repo" "${projectBasePath}/java/${3}/${1}/domain/dao" "${projectBasePath}/java/${3}/${1}/domain/model"
+  mkdir -p "${projectBasePath}/java/${3}/${1}/logic" "${projectBasePath}/java/${3}/${1}/rest/v1/mapper" "${projectBasePath}/java/${3}/${1}/rest/v1/model"
+  mkdir -p "${projectBasePath}/java/${3}/${1}/rest/v1"
```

Comment:
we could also consider providing a template as archive (ZIP or TGZ) that is downloaded and extracted on top to make this easier to maintain. That archive could even be autogenerated from a folder-structure in git.
However, just a nice-to-have idea. Maybe we should rather create a backlog story for this and not prevent this PR getting merged with it.

Reply:
IMHO this would even belong to devon4quarkus repo.

Reply:
That's a good idea. Where could we put the zip then?